processing/data_collection/gazette/spiders/sp_santo_andre.py

```diff
@@ -0,0 +1,187 @@
+import dateparser
+from datetime import datetime, date
+from urllib.parse import unquote
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+from scrapy import FormRequest, Request
+from scrapy.selector.unified import Selector
+from gazette.settings import FILES_STORE
+import re
+from w3lib.html import remove_tags
+from pathlib import Path
+from typing import Optional
+from hashlib import sha1
+
+
+class SpSantoAndreSpider(BaseGazetteSpider):
+    name = "sp_santo_andre"
+    TERRITORY_ID = "3547809"
+    custom_settings = {
+        "ITEM_PIPELINES": {
+            "gazette.pipelines.GazetteDateFilteringPipeline": 50,
+            "gazette.pipelines.ExtractTextPipeline": 200,
+        }
+    }
+
+    allowed_domains = ["santoandre.sp.gov.br"]
+    start_urls = [
+        "http://www.santoandre.sp.gov.br/publicacao/edicao/consultaedicao.aspx"
+    ]
+    start_date = date(2010, 1, 1)
+    TODAY = datetime.now().strftime("%d/%m/%Y")
+    JAVASCRIPT_POSTBACK_REGEX = r"javascript:__doPostBack\('(.*)',''\)"
+    CURRENT_PAGE = 1
+
+    def start_requests(self):
+        for url in self.start_urls:
+            yield Request(url, callback=self.search_by_date)
+
+    def _get_form_params(self, response):
+        view_state = response.css("#__VIEWSTATE::attr(value)").get()
+        event_validation = response.css("#__EVENTVALIDATION::attr(value)").get()
+        return view_state, event_validation
+
+    def search_by_date(self, response):
+        VIEW_STATE, EVENT_VALIDATION = self._get_form_params(response)
+        yield FormRequest.from_response(
+            response,
+            callback=self.parse,
+            formname="aspnetForm",
+            formdata={
+                "__EVENTTARGET": "",
+                "__EVENTARGUMENT": "",
+                "__VIEWSTATE": VIEW_STATE,
+                "__EVENTVALIDATION": EVENT_VALIDATION,
+                "ctl00$ContentPlaceHolder1$dt_inicio": self.start_date.strftime(
+                    "%d/%m/%Y"
+                ),
+                "ctl00$ContentPlaceHolder1$dt_final": self.TODAY,
+            },
+            clickdata={"name": "ctl00$ContentPlaceHolder1$PsaToolBar1$btnSelecionar"},
+            method="POST",
+        )
+
+    def _save_pdf(self, response, path: Path, item):
```

Comment:
**suggestions**: We don't use type hints on the spiders. To keep all spiders consistent it would be better to remove it.

Reply:
OK! I'm very susceptible to use typing annotations these days. But I understand that we need to promote a coding standard for the project.